actors/common/src/main/java/org/sunbird/validator/user/UserBulkMigrationRequestValidator.java

```diff
@@ -0,0 +1,200 @@
+                    }
+                }
+        );
+        }
+
+    private void supportedColumns(){
+        migration.getHeaders().forEach(suppColumn->{
+            if(!migration.getSupportedFields().contains(suppColumn.toLowerCase())){
+                ProjectLogger.log("UserBulkMigrationRequestValidator:supportedColumns: supported column is not present".concat(suppColumn+""), LoggerEnum.ERROR.name());
+                throw new ProjectCommonException(
+                        ResponseCode.errorUnsupportedField.getErrorCode(),
+                        ResponseCode.errorUnsupportedField.getErrorMessage(),
+                        ResponseCode.CLIENT_ERROR.getResponseCode(),
+                        "Invalid provided column ".concat(suppColumn).concat("supported headers are:").concat(ArrayUtils.stringifyContents(migration.getSupportedFields().toArray())));
+            }
+        });
+    }
+
+
+
+    private void csvRows(){
+        validateRowsCount();
+        migration.getValues().stream().forEach(migrationUser -> {
+            int index=migration.getValues().indexOf(migrationUser);
+            validateMigrationUser(migrationUser,index);
+        });
+        if(csvRowsErrors.getErrorsList().size()>0){
+            IErrorDispatcher errorDispatcher= ErrorDispatcherFactory.getErrorDispatcher(csvRowsErrors);
+            errorDispatcher.dispatchError();
+        }
+    }
+
+    private void validateRowsCount(){
+        if(migration.getValues().size()>=MAX_ROW_SUPPORTED){
+            throw new ProjectCommonException(
+                    ResponseCode.csvRowsExceeds.getErrorCode(),
+                    ResponseCode.csvRowsExceeds.getErrorMessage().concat("supported:"+MAX_ROW_SUPPORTED),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+        }
+    }
+    private void validateMigrationUser(MigrationUser migrationUser,int index) {
+      emailAndPhone(migrationUser.getEmail(),migrationUser.getPhone(),index);
+      userExternalId(migrationUser.getUserExternalId(),index);
+      name(migrationUser.getName(),index);
+      orgExternalId(migrationUser.getOrgExternalId(),index);
+      channel(migrationUser.getChannel(),index);
+      inputStatus(migrationUser.getInputStatus(),index);
+    }
+    private void addErrorToList(ErrorDetails errorDetails){
+        if(errorDetails.getErrorEnum()!=null){
+            csvRowsErrors.setError(errorDetails);
+        }
+    }
+
+    public void emailAndPhone(String email, String phone,int index) {
+        ErrorDetails errorDetails=new ErrorDetails();
+        errorDetails.setRowId(index);
+        if(StringUtils.isBlank(email) && StringUtils.isBlank(phone)){
+            errorDetails.setErrorEnum(ErrorEnum.missing);
+            errorDetails.setHeader(JsonKey.EMAIL);
+        }
+        else if(StringUtils.isNotBlank(email)){
+            errorDetails.setHeader(JsonKey.EMAIL);
+            boolean isEmailValid=ProjectUtil.isEmailvalid(email);
+            if(!isEmailValid){
```

Comment:
if email is not valid then we don't need to check for duplicate, because anyways user need to change it.
same applicable for phone as well.

Reply:
done